store/root/store.go

```diff
@@ -5,9 +5,12 @@ import (
 	"fmt"
 	"io"
 	"slices"
+	"time"
 
 	"github.com/cockroachdb/errors"
 
+	"github.com/cosmos/cosmos-sdk/telemetry"
```

Comment:
would be good to avoid importing the sdk. This will allow store to be used across versions without needing to be upgraded

Reply:
I don't see how we can? The `telemetry` package does all the telemetry logic such as setting up the global labels and creating the Prometheus sink (see [here](https://github.com/cosmos/cosmos-sdk/blob/main/telemetry/metrics.go#L74-L122)).

Unless you want to copy pasta all the code, I don't see how we can avoid using it. We also can't use an interface, because APIs such as `MeasureSince` are functions, not methods on a type.

Why not just make `telemetry` a standalone package?

Reply:
https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/store/metrics/telemetry.go

this is what we did for store v1. 

i added the noop but never got around to wiring based off prometheus being turned on or off. I dont see a need for it since it will most likely always be turned on. If we do see a performance overhead we can have a developer mode for metrics and a production mode.

Reply:
Yeah, this is what I originally had (see: https://github.com/cosmos/cosmos-sdk/pull/18529/commits/96170327f1fb7168c25410317a998fd03643b690). But I opted out of this because it doesn't share the global labels that's set in the `telemetry` package.

After digging through it again, I can now see the magic here:

```go
	if _, err := metrics.NewGlobal(metricsConf, fanout); err != nil {
		return nil, err
	}
```

So using this approach should work. I'll update.